src/components/InventoryGroups/Modals/AddSystemsToGroupModal.js

```diff
@@ -92,14 +111,75 @@ const AddSystemsToGroupModal = ({
     dispatch(clearFilters());
   };
 
+  const edgeParityEnabled =
+    edgeParityIsAllowed &&
+    useFeatureFlag('edgeParity.inventory-list') &&
+    useFeatureFlag('edgeParity.inventory-groups-enabled');
+
+  const [selectedImmutableDevices, setSelectedImmutableDevices] = useState([]);
+  const selectedImmutableKeys = selectedImmutableDevices.map(
+    (immutableDevice) => immutableDevice.id
+  );
+
+  // overallSelectedKeys is the list of the conventional and immutable systems ids
+  const overallSelectedKeys = [...selected.keys(), ...selectedImmutableKeys];
+  // noneSelected a boolean showing that no system is selected
+  const noneSelected = overallSelectedKeys.length === 0;
+
+  const immutableDevicesAlreadyHasGroup = selectedImmutableDevices.filter(
+    (immutableDevice) => immutableDevice.deviceGroups?.length > 0
+  );
+  // show warning also when immutable systems had groups
+  showWarning = showWarning || immutableDevicesAlreadyHasGroup.length > 0;
```

Comment:
It might be good to wrap this into a `useMemo` and also avoid reassigning `showWarning` from line 77. Things like these have caused unexpected behaviour with React.

Reply:
Thanks for pointing this,  will assign it only once then 
```js
const showWarning = alreadyHasGroup.length > 0 || immutableDevicesAlreadyHasGroup.length > 0;
```